pkg/util/http.go

```diff
@@ -122,45 +148,72 @@ func ParseProtoReader(ctx context.Context, reader io.Reader, expectedSize, maxSi
 		// reader is over limit, the result will be bigger than max.
 		_, err = buf.ReadFrom(io.LimitReader(reader, int64(maxSize)+1))
 		body = buf.Bytes()
-	case FramedSnappy:
-		_, err = buf.ReadFrom(io.LimitReader(snappy.NewReader(reader), int64(maxSize)+1))
-		body = buf.Bytes()
 	case RawSnappy:
 		_, err = buf.ReadFrom(reader)
 		body = buf.Bytes()
 		if sp != nil {
 			sp.LogFields(otlog.String("event", "util.ParseProtoRequest[decompress]"),
 				otlog.Int("size", len(body)))
 		}
-		if err == nil && len(body) <= maxSize {
+		if err != nil {
+			return nil, err
+		}
+		if len(body) > maxSize {
```

Comment:
Why don't we call `decompressFromBuffer()` instead of the code below? Looks doing the same thing.

Reply:
Good feedback, refactored a bit and made the change.

Reply:
I didn't had tests before and I was reluctant to change too much, but now I do have tests and they still all pass.